azure/types.go

```diff
@@ -67,7 +67,7 @@ type ScaleSetSpec struct {
 	DiagnosticsProfile           *infrav1.Diagnostics
 	FailureDomains               []string
 	VMExtensions                 []infrav1.VMExtension
-	NetworkInterfaces            []infrav1.NetworkInterface
+	NetworkInterfaces            []infrav1.MachinePoolNetworkInterface
```

Comment:
APIDiff test is complaining about this change.
Shall we be just calling it out in release notes about this change or is there more we can do about it? https://github.com/everyone.

Reply:
This is an interesting one because it is a false-positive from the perspective of the JSON/YAML schema. I asked this during office hours and @mboersma mentioned those types of apidiff false-positives happen from time to time. Only the struct name changed, but no field names have changed. I was also told the apidiff test is not required for PRs for this reason.

Reply:
^ thanks for the context, Dane! Makes sense!